src/storage-controller/src/collection_mgmt.rs

```diff
@@ -610,6 +671,18 @@ where
 
         self.write_to_persist(responders).await?;
 
+        // Reset the interval which is used to periodically bump the uppers
+        // because the uppers will get bumped with the following update.
+        // This makes it such that we will write at most once every
+        // `interval`.
+        //
+        // For example, let's say our `DEFAULT_TICK` interval is 10, so at
+        // `t + 10`, `t + 20`, ... we'll bump the uppers. If we receive an
+        // update at `t + 3` we want to shift this window so we bump the
+        // uppers at `t + 13`, `t + 23`, ... which resetting the interval
+        // accomplishes.
+        self.upper_tick_interval.reset();
```

Comment:
True, I've been moving it closer to the write, because I had it in the if block. I'll change it back around to the previous behavior of resetting before the write 👌